packages/block-library/src/button/style.scss

```
@@ -50,58 +49,35 @@ $blocks-block__margin: 0.5em;
 	}
 
 	&.wp-block-button__width-25 {
-		width: calc(25% - #{ $blocks-block__margin });
+		width: calc(25% - (var(--wp--style--block-gap, $blocks-block__margin) * 0.75));
```

It looks like to render the Sass variable `$blocks-block__margin` in this context (nested within var within a calc) we need to wrap it in the [Sass interpolation syntax](https://sass-lang.com/documentation/breaking-changes/css-vars) where the variable is wrapped in `#{}`, so for example this line would become:

```suggestion
		width: calc(25% - (var(--wp--style--block-gap, #{$blocks-block__margin}) * 0.75));
```

---

Thanks as always for the thorough testing -- I should've tested on a non-blocks theme! Updated 💯